File: App/src/Adaptive.ReactiveTrader.Client/UI/SpotTiles/IOneWayPriceViewModel.cs
using System.Windows.Input;
using Adaptive.ReactiveTrader.Client.Models;

namespace Adaptive.ReactiveTrader.Client.UI.SpotTiles
{
    public interface IOneWayPriceViewModel : IViewModel
    {
        Direction Direction { get; }
        string BigFigures { get; }
        string Pips { get; }
        string TenthOfPip { get; }
        PriceMovement Movement { get; }
        ICommand ExecuteCommand { get; }
        void OnPrice(IExecutablePrice executablePrice);
        void OnStalePrice();
    }
}

File: App/src/Adaptive.ReactiveTrader.Client.JS/UI/IOneWayPriceViewModel.ts
interface IOneWayPriceViewModel {
    onExecute(): void;
    onPrice(executablePrice: IExecutablePrice): void;
    onStalePrice(): void;

    direction: string;
    bigFigures: KnockoutObservable<string>;
    pips: KnockoutObservable<string>;
    tenthOfPips: KnockoutObservable<string>;
    isExecuting: KnockoutObservable<boolean>;
    isStale: KnockoutObservable<boolean>;
} 